.github/workflows.md

```diff
@@ -0,0 +1,70 @@
+# Overview
+
+This document explains how and why we organize our workflows for a CI strategy using GitHub actions.
+
+## Directories and file structure
+
+Our directory structure is as follows:
+
+- 📁 `/actions` contains reusable actions needed in workflows
+- 📁 `/workflows` contains GitHub workflow files
+- 📁 `/ISSUES_TEMPLATE` contains  templates for our issues
+
+We have two types of files in the `/workflows` directory:
+-  📄 starting with `job_` are reusable jobs that are used in different workflows.
+  - they should (and can't) be used in isolation, as they don't take care of the code being checked out
+-  📄 starting with `workflow_` indicate a collection of jobs that are automatically triggered by certain events or actions
+
+## Pipelines 
+
+We have two types of workflows triggered by Pull Requests:
+
+1. When a Pull Request is opened, we run selected jobs based on the location and changes made in the code. We have two sets of jobs:
+   - if changes are made inside the `/coral` directory, we run all [coral jobs](./workflows/jobs-coral.yaml)
+   - if are made outside in `/core` or `/cluster-api` directories or `pom.xml`, we run all [maven jobs](./workflows/jobs-maven.yaml). This jobs are only run when the Pull Request is ready for review, excluding "Draft" PRs.
+   
+2. When a Pull Request is approved, the [`workflow-merge-to-main`](./workflows/workflow-merge-to-main.yaml) is triggered.
+   - This workflow runs all [coral jobs](./workflows/jobs-coral.yaml) and [maven jobs](./workflows/jobs-maven.yaml)
+   - After they are successful, it runs a job called `merge-to-main`. This job is defined in the [Branch protection rule](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule)  for `main` and is and is necessary to enable merging on our default branch.
+
+In addition, we have a workflow that is triggered when changes are made to the `openapi.yaml file`. We  check whether the changes affect the TypeScript file in `/coral`. If the changes do affect the TypeScript file, the `api.d.ts` file is automatically generated, and the TypeScript compiler is run. If the TypeScript compiler finds no errors, the file is committed. Otherwise the job fails.
+
+## Drawbacks
+
+We have identified two potential drawbacks with this strategy:
+
+1. There will be a delay between approving a Pull Request and merging it because of the required job on approval. We can mitigate this in the future:
+   - by adding automatic notification about a PR being ready to merge. 
+   - consider auto-merging it in the future. Currently, we prefer manual merges to maintain more control over the process. 
```

Comment:
"Merge when ready" is a merge queue feature, right? Probably what we would like ideally?

Reply:
Yes. I mean I'm not sure how it's called, but merge queues should enable us to -> press merge on a branch when it's approved and the pull-request related pipelines have run -> then a temporary branch is created where the now merge-to-main pipeline is run -> if that's successful it will get automatically merged into main, if not, we get a notification etc. 

My last information (based on github docs etc) was that merge queues are still public beta and prone to changes and a few issues I read up on this struggled with some things, so I decided against using them for Klaw right now, but would look into it when it looks stable enough to be used by us as a small team.